pkg/controller/pod_test.go

```diff
@@ -301,6 +303,58 @@ var _ = Describe("Dynamic Attachment controller", func() {
 				})
 			})
 
+			When("an attachment is removed and another is added from the pod's network annotations", func() {
```

Comment:
these tests look really good, and align with the rest of the test suite ! 

Thanks for that.